libs/coin-modules/coin-evm/src/api/node/ledger.ts

```diff
@@ -395,6 +394,72 @@ export const getOptimismAdditionalFees: NodeApi["getOptimismAdditionalFees"] = a
   return new BigNumber(result.response);
 };
 
+/**
+ * ⚠️ Blockchain specific
+ *
+ * For a layer 2 like Scroll, additional fees are needed in order to
+ * take into account layer 1 settlement estimated cost.
+ * This gas price is served through a smart contract oracle.
+ *
+ * @see https://docs.scroll.io/en/developers/transaction-fees-on-scroll/
+ */
+export const getScrollAdditionalFees: NodeApi["getScrollAdditionalFees"] = async (
+  currency,
+  transaction,
+) => {
+  const config = getCoinConfig(currency).info;
+  const { node } = config || /* istanbul ignore next */ {};
+  if (!isLedgerNodeConfig(node)) {
+    throw new LedgerNodeUsedIncorrectly();
+  }
+
+  if (!["scroll", "scroll_sepolia"].includes(currency.id)) {
+    return new BigNumber(0);
```

Comment:
Shouldn't it throw an error if we can call this function from an unsupported currency ?

Reply:
We could, I just prefer a graceful fallback when possible to avoid having huge errors during development for a function called during the `prepareTransaction` phase.